### PR TITLE
Cleaning up clip and transform

### PR DESCRIPTION
**Pull request by jherico**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: a50097997e41019ed3b6c4905526540e9c267712 available at: https://gist.github.com/ef9403c6bef89ec50d5d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6377.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6377.dmg

File: libraries/shared/src/Transform.cpp [hunk 1]
```diff
@@ -65,7 +71,70 @@ void Transform::evalRotationScale(Quat& rotation, Vec3& scale, const Mat3& rotat
...
+Transform Transform::worldTransform(const Transform& relativeTransform) const {
+    if (relativeTransform.isIdentity()) {
+        return *this;
+    }
 
+    if (isIdentity()) {
+        return relativeTransform;
+    }
 
+    return Transform(getMatrix() * relativeTransform.getMatrix());
```
Comment (samcake) [inline libraries/shared/src/Transform.cpp]: You don't have to go back to Matrix to perform the transformation
You can just do:
return mult(Transform(), (*this), relativeTransform);

File: libraries/shared/src/Transform.cpp [hunk 2]
```diff
@@ -65,7 +71,70 @@ void Transform::evalRotationScale(Quat& rotation, Vec3& scale, const Mat3& rotat
...
+Transform Transform::relativeTransform(const Transform& worldTransform) const {
+    if (isIdentity()) {
+        return worldTransform;
+    }
+
+    if (*this == worldTransform) {
+        return Transform();
+    }
+
+    return Transform(getInverseMatrix() * worldTransform.getMatrix());
```
Comment (samcake) [inline libraries/shared/src/Transform.cpp]: You don't have to go back to Matrix to perform the transformation
You can just do:
return inverseMult(Transform(), (*this), worldTransform);

File: libraries/shared/src/Transform.cpp (hunk 1 above)
Comment (jherico) [inline libraries/shared/src/Transform.cpp]: you can't pass a temp into a non-const reference, but I will adapt this.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ee1545f649a254dba61ac3e2a1c7156365a45b25 available at: https://gist.github.com/ef9403c6bef89ec50d5d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6377.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6377.dmg

**Pull request merged**